### PR TITLE
Add an API for getting the CFL number.

### PR DESCRIPTION
**Pull request by drwells**
Presently, the CFL number is only available through the logfile. This PR consolidates where we calculate it (the INS solver is always responsible) and makes it available via `getCurrentCFLNumber()`.

### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Review by abarret — APPROVED** [review]
A couple of comments.

File: include/ibamr/INSHierarchyIntegrator.h [hunk 1]
```diff
@@ -422,6 +434,12 @@ class INSHierarchyIntegrator : public IBTK::HierarchyIntegrator
      */
     virtual void regridProjection() = 0;
 
+    /*!
+     * Pure virtual method to update the current CFL number (i.e., at the end of
+     * a timestep).
+     */
+    virtual void updateCurrentCFLNumber(const int data_idx, const double dt) = 0;
```
Comment (boyceg) [inline include/ibamr/INSHierarchyIntegrator.h:441]: I wonder about making this a static that "knows what to do" for different data centerings.

I guess the advantage of making it virtual is if we ever get around to implementing INS schemes that use non-interpolatory kernels.

Note also that a generic implementation could be provided that uses functionality from `ArrayDataNormOps` to do this in a somewhat data-centering agnostic way.

Comment (drwells) [reply include/ibamr/INSHierarchyIntegrator.h:441]: We could make it static but I think a good alternative would be to recycle the external version in IBHierarchyIntegrator which can handle both data centerings.

Data is stored in a weird way for side-centered so I don't think using the array operations would make this significantly simpler.

File: include/ibamr/INSHierarchyIntegrator.h [hunk 2]
```diff
@@ -371,6 +371,18 @@ class INSHierarchyIntegrator : public IBTK::HierarchyIntegrator
...
+                                               double new_time,
+                                               bool skip_synchronize_new_state_data,
+                                               int num_cycles = 1) override;
+
+    /*!
+     * Return the current CFL number (i.e., the CFL number computed from the
+     * current velocity field). This is typically computed at the end of each
+     * time step.
+     */
+    virtual double getCurrentCFLNumber() const;
```
[the remark this answers is not in the document]
Comment (drwells) [reply include/ibamr/INSHierarchyIntegrator.h:384]: No, but it can't hurt - maybe someone will do something funny to compute this in the future.

**Pull request merged**